From f82d437499a10686227ed8e4416d1267a5de1fb7 Mon Sep 17 00:00:00 2001
From: Eberhard Beilharz <eb1@sil.org>
Date: Mon, 30 Apr 2012 16:26:25 +0200
Subject: [PATCH 2/2] Xamarin-4739: Fix selection of child nodes

XNodeNavigator.Select() missed child nodes that were
surrounded by text.
---
 .../System.Xml.Linq/XNodeNavigator.cs              |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: fieldworks-mono/mcs/class/System.Xml.Linq/System.Xml.Linq/XNodeNavigator.cs
===================================================================
--- fieldworks-mono.orig/mcs/class/System.Xml.Linq/System.Xml.Linq/XNodeNavigator.cs
+++ fieldworks-mono/mcs/class/System.Xml.Linq/System.Xml.Linq/XNodeNavigator.cs
@@ -336,7 +336,7 @@
 		public override bool MoveToNext ()
 		{
 			XNode xn = node.NextNode;
-			if (node is XText)
+			if (xn is XText)
 				for (; xn != null; xn = xn.NextNode)
 					if (!(xn.NextNode is XText))
 						break;
